controllers/reconcilers/logging_installation/logging_installation_reconciler.go

```diff
@@ -0,0 +1,298 @@
+	}
+
+	var foundLoggingOperator = false
+
+	for _, deployment := range deployments.Items {
+		if strings.HasPrefix(deployment.Name, "cluster-logging-operator") {
+			foundLoggingOperator = true
+		}
+	}
+
+	// If there was no logging-operator and we want to install the operator go ahead and set it up
+	if !foundLoggingOperator {
+		// logging subscription
+		status, err := r.reconcileSubscription(ctx, cr)
+		if status != v1.ResultSuccess {
+			return status, err
+		}
+
+		status, err = r.waitForLoggingOperator(ctx, cr)
+		if status != v1.ResultSuccess {
+			return status, err
+		}
+	}
+	
+	status, err := r.createClusterLoggingCr(ctx, cr)
+	if status != v1.ResultSuccess {
+		return status, err
+	}
+
+	status, err = r.createClusterLogForwarderCr(ctx, cr)
+	if status != v1.ResultSuccess {
+		return status, err
+	}
+
+	return v1.ResultSuccess, nil
+}
+
+func (r *Reconciler) reconcileSubscription(ctx context.Context, cr *v1.Observability) (v1.ObservabilityStageStatus, error) {
+
+	subscription := model.GetLoggingSubscription(cr)
+
+	_, err := controllerutil.CreateOrUpdate(ctx, r.client, subscription, func() error {
+		subscription.Spec = &v1alpha1.SubscriptionSpec{
+			CatalogSource:          "redhat-operators",
+			CatalogSourceNamespace: "openshift-marketplace",
+			Package:                "cluster-logging",
+			Channel:                "stable",
+			InstallPlanApproval:    v1alpha1.ApprovalAutomatic,
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return v1.ResultFailed, err
+	}
+
+	return v1.ResultSuccess, nil
+}
+
+func (r *Reconciler) waitForLoggingOperator(ctx context.Context, cr *v1.Observability) (v1.ObservabilityStageStatus, error) {
+	deployments := &v12.DeploymentList{}
+	opts := &client.ListOptions{
+		Namespace: "openshift-logging",
+	}
+	err := r.client.List(ctx, deployments, opts)
+	if err != nil {
+		return v1.ResultFailed, err
+	}
+
+	for _, deployment := range deployments.Items {
+		if strings.HasPrefix(deployment.Name, "cluster-logging-operator") {
+			if deployment.Status.ReadyReplicas > 0 {
+				return v1.ResultSuccess, nil
+			}
+		}
+	}
+	return v1.ResultInProgress, nil
+}
+
+func (r *Reconciler) createClusterLoggingCr(ctx context.Context, cr *v1.Observability) (v1.ObservabilityStageStatus, error) {
+
+
+	// Is there any clusterLogging CR?
+	opts := &client.ListOptions{
+		Namespace: "openshift-logging",
+	}
+
+	list := &v14.ClusterLoggingList{}
+	err := r.client.List(ctx, list, opts)
+	if err != nil {
+		return v1.ResultFailed, err
+	}
+
+	// Is there a clusterLogging CR with our label?
+	OOopts := &client.ListOptions{
+		Namespace: "openshift-logging",
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			"app.kubernetes.io/managed-by": "observability-operator",
+		}),
+	}
+	OOlist := &v14.ClusterLoggingList{}
+	err = r.client.List(ctx, OOlist, OOopts)
+	if err != nil {
+		return v1.ResultFailed, err
+	}
+
+
+	if len(list.Items) == 0 || len(OOlist.Items) == 0 || len(OOlist.Items) > 0 {
```

Comment:
isn't this condition always true? OOList has a length of either zero or greater zero.

Reply:
Yes you are correct, I added that so it would just always run in testing, I've fixed it it now.